### ISSUE TITLE
Getting error in configuring default binder while configuring with Multiple binders

### ISSUE THREAD

**Issue by sabbyanandan** [spring-cloud/spring-cloud-stream-binder-kafka#793]
@muhammad-usman-confiz commented on [Fri Nov 01 2019](https://github.com/spring-cloud/spring-cloud-stream/issues/1839)

Using `spring cloud stream` binders, I am trying to connect with multiple azure event hub namespaces. This configuration works fine with single namespace but for multiple namespaces, I am getting following exception on startup.

```
Caused by: java.lang.IllegalStateException: A default binder has been requested, but there is more than one binder available for 'org.springframework.cloud.stream.messaging.DirectWithAttributesChannel' : kafka2,kafka1, and no default binder has been set.
	at org.springframework.cloud.stream.binder.DefaultBinderFactory.doGetBinder(DefaultBinderFactory.java:181) ~[spring-cloud-stream-3.0.0.RC1.jar:3.0.0.RC1]
	at org.springframework.cloud.stream.binder.DefaultBinderFactory.getBinder(DefaultBinderFactory.java:132) ~[spring-cloud-stream-3.0.0.RC1.jar:3.0.0.RC1]
	at org.springframework.cloud.stream.binding.BindingService.getBinder(BindingService.java:339) ~[spring-cloud-stream-3.0.0.RC1.jar:3.0.0.RC1]
	at org.springframework.cloud.stream.binding.BindingService.bindProducer(BindingService.java:234) ~[spring-cloud-stream-3.0.0.RC1.jar:3.0.0.RC1]
	at org.springframework.cloud.stream.binding.AbstractBindableProxyFactory.createAndBindOutputs(AbstractBindableProxyFactory.java:136) ~[spring-cloud-stream-3.0.0.RC1.jar:3.0.0.RC1]
	at org.springframework.cloud.stream.binding.OutputBindingLifecycle.doStartWithBindable(OutputBindingLifecycle.java:58) ~[spring-cloud-stream-3.0.0.RC1.jar:3.0.0.RC1]
	at java.util.LinkedHashMap$LinkedValues.forEach(LinkedHashMap.java:608) ~[na:1.8.0_221]
	at org.springframework.cloud.stream.binding.AbstractBindingLifecycle.start(AbstractBindingLifecycle.java:57) ~[spring-cloud-stream-3.0.0.RC1.jar:3.0.0.RC1]
	at org.springframework.cloud.stream.binding.OutputBindingLifecycle.start(OutputBindingLifecycle.java:34) ~[spring-cloud-stream-3.0.0.RC1.jar:3.0.0.RC1]
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:182) ~[spring-context-5.2.0.RELEASE.jar:5.2.0.RELEASE]
	... 14 common frames omitted
```

Here is my .yml configuration

```
spring:
  cloud:
    stream:
      kafka:
        binder:
          brokers: broker1.servicebus.windows.net:9093
          configuration:
            sasl.jaas.config: "..."
            sasl.mechanism: PLAIN
            security.protocol: SASL_SSL
      dynamicDestinations: dev-event-hub,prod-event-hub
      bindings:
        input-1:
          destination: dev-event-hub,prod-event-hub
          binder: kafka1
          group: $Default
        input-2:
          destination: dev-event-hub,prod-event-hub
          binder: kafka2
          group: $Default
      binders:
        kafka1:
          type: kafka
          environment:
            spring:
              cloud:
                stream:
                  Kafka:
                    binder:
                      brokers: broker2.servicebus.windows.net:9093
                      configuration:
                        sasl.jaas.config: "..."
                        sasl.mechanism: PLAIN
                        security.protocol: SASL_SSL
                  dynamicDestinations: dev-event-hub,prod-event-hub
        kafka2:
          type: kafka
          environment:
            spring:
              cloud:
                stream:
                  Kafka:
                    binder:
                      brokers: broker3.servicebus.windows.net:9093
                      configuration:
                        sasl.jaas.config: "..."
                        sasl.mechanism: PLAIN
                        security.protocol: SASL_SSL
                  dynamicDestinations: dev-event-hub,prod-event-hub
```

**Comment by sobychacko** [comment]
@muhammad-usman-confiz Your configuration looks right. Can you remove that initial configuration for binders where you define the brokers since you already provide that as part of multi-binder configuration? If it still doesn't work, please provide a small sample for us to test and make the brokers local in the sample. Did you try [this](https://github.com/spring-cloud/spring-cloud-stream-samples/tree/master/multi-binder-samples/multi-binder-two-kafka-clusters) sample?

**Comment by muhammad-usman-confiz** [comment]
@sobychacko Due to project deadline I used various other ways to connect with multiple event hub namespaces (multiple Kafka clusters) and ended up on `spring-Kafka` with totally customized solutions. 

After your reply, I tried again using the `spring-cloud-stream` to reproduce this issue but now without default Kafka binder, so this solution is working fine. 

I think I should explore more on my side whether this issue was due to my configuration or any other reason.

Thanks
